File: apps/backend/internal/requests/domain/entity/service_request.go
package entity

import "time"

type ServiceRequest struct {
	ID            string     `json:"id"`
	ClientID      string     `json:"client_id"`
	CompanionID   string     `json:"companion_id"`
	ServiceType   string     `json:"service_type"`
	FullName      string     `json:"full_name"`
	Phone         string     `json:"phone"`
	Address       string     `json:"address"`
	MeetingPoint  *string    `json:"meeting_point,omitempty"`
	PreferredDate string     `json:"preferred_date"`
	Notes         *string    `json:"notes,omitempty"`
	Status        string     `json:"status"`
	ExpiresAt     *time.Time `json:"expires_at,omitempty"`
	CreatedAt     time.Time  `json:"created_at"`
	UpdatedAt     time.Time  `json:"updated_at"`
}

File: apps/legacy/mobile-flutter/lib/features/client/domain/entities/service_request.dart
import 'request_status.dart';

class ServiceRequest {
  final String id;
  final String clientId;
  final String? companionId;
  final String serviceType;
  final String fullName;
  final String phone;
  final String? address;
  final String? meetingPoint;
  final String preferredDate;
  final String? notes;
  final RequestStatus status;
  final DateTime createdAt;
  final DateTime? updatedAt;

  const ServiceRequest({
    required this.id,
    required this.clientId,
    this.companionId,
    required this.serviceType,
    required this.fullName,
    required this.phone,
    this.address,
    this.meetingPoint,
    this.preferredDate = '',
    this.notes,
    this.status = RequestStatus.pending,
    required this.createdAt,
    this.updatedAt,
  });
}
